accounting/job_archive_interface.py

```diff
@@ -218,3 +220,129 @@ def append_to_where(where_stmt, conditional):
             write_records_to_file(job_records, output_file)
 
     return job_records
+
+
+def apply_decay_factor(decay_factor, acct_conn, user=None, bank=None):
+    past_usage_factors = []
+    past_usage_factors_w_decay = []
+
+    select_stmt = "SELECT * from job_usage_factor_table WHERE username=? AND bank=?"
+    dataframe = pd.read_sql_query(select_stmt, acct_conn, params=(user, bank,))
+
+    for val in dataframe.iloc[0].values:
+        if isinstance(val, float):
+            past_usage_factors.append(val)
+
+    # apply decay factor to past usage periods of a user's jobs
+    power = 1
+    for usage_factor in past_usage_factors:
+        past_usage_factors_w_decay.append(usage_factor * math.pow(decay_factor, power))
+        power += 1
```

Comment:
Totally fine to leave this code as is. Just wanted to share a neat little trick I learned a while back:

```python
for power, usage_factor in enumerate(past_usage_factors, start=1):
	past_usage_factors_w_decay.append(usage_factor * math.pow(decay_factor, power))
```

https://docs.python.org/3/library/functions.html#enumerate